src/rpc/blockchain.cpp

```diff
@@ -2284,6 +2284,108 @@ static RPCHelpMan scantxoutset()
     };
 }
 
+
+static RPCHelpMan scanblockfilters()
+{
+    return RPCHelpMan{"scanblockfilters",
+                "\nReturn relevant blockhashes for given descriptors.\n",
+                {
+                    {"scanobjects", RPCArg::Type::ARR, RPCArg::Optional::NO, "Array of scan objects.\n"
+            "                                  Every scan object is either a string descriptor or an object:",
+                        {
+                            {"descriptor", RPCArg::Type::STR, RPCArg::Optional::OMITTED, "An output descriptor"},
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "An object with output descriptor and metadata",
+                                {
+                                    {"desc", RPCArg::Type::STR, RPCArg::Optional::NO, "An output descriptor"},
+                                    {"range", RPCArg::Type::RANGE, /* default */ "1000", "The range of HD chain indexes to explore (either end or [begin,end])"},
+                                },
+                            },
+                        },
+                        "[scanobjects,...]"},
+                    {"startheight", RPCArg::Type::NUM, /*default*/ "0", "height to start to filter from"},
+                    {"filtertype", RPCArg::Type::STR, /*default*/ "basic", "The type name of the filter"}
+                },
+                RPCResult{
+                    RPCResult::Type::ARR, "", "",
+                    {
+                        {RPCResult::Type::STR_HEX, "", "The blockhash"},
+                    }},
+                RPCExamples{
+                    HelpExampleCli("scanblockfilters", "[\"addr(bcrt1q4u4nsgk6ug0sqz7r3rj9tykjxrsl0yy4d0wwte)\"] 300000") +
+                    HelpExampleRpc("scanblockfilters", "[\"addr(bcrt1q4u4nsgk6ug0sqz7r3rj9tykjxrsl0yy4d0wwte)\"] 300000")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::string filtertype_name = "basic";
+    if (!request.params[2].isNull()) {
+        filtertype_name = request.params[1].get_str();
```

Comment:
Looks like this should be `params[2]`
```suggestion
        filtertype_name = request.params[2].get_str();
```

In the functional test, suggest asserting that two calls like this return the same result:
```
bitcoin-cli -signet scanblockfilters \
  '["addr(tb1qdydhyacp7tj2asc0u4mhtf6rn65dfjxclrdd7a)"]' 0 basic
[
  "0000013b318daf6e8d4c531b8293ebf800871f4220fc4b7b25cc6798ec5cb155",
  "000001003f625cbecd53924713929b5a91b12edf4eb844f33781ecb6bb78e69b",
  "000000840904ea8ff963d4f776c0175a622228738cdf9ebd481e3ef483ccdff0"
]

bitcoin-cli -signet scanblockfilters \
  '["addr(tb1qdydhyacp7tj2asc0u4mhtf6rn65dfjxclrdd7a)"]' 0 
[
  "0000013b318daf6e8d4c531b8293ebf800871f4220fc4b7b25cc6798ec5cb155",
  "000001003f625cbecd53924713929b5a91b12edf4eb844f33781ecb6bb78e69b",
  "000000840904ea8ff963d4f776c0175a622228738cdf9ebd481e3ef483ccdff0"
]
```

Reply:
Can be `const`
```diff
- std::string filtertype_name = "basic";
- if (!request.params[2].isNull()) {
-     filtertype_name = request.params[1].get_str();
- }
+ const std::string filtertype_name{request.params[2].isNull() ? "basic" : request.params[2].get_str()};
```

Reply:
Nice. Will adapt this.